services/black/src/api/RequestGuestInformationHandler.cs

```diff
@@ -1,16 +1,28 @@
 ﻿using System.Threading.Tasks;
+using artiso.AdsdHotel.Black.Api.DatabaseModel;
 using artiso.AdsdHotel.Black.Commands;
-using artiso.AdsdHotel.Black.Contracts;
+using artiso.AdsdHotel.Infrastructure.DataStorage;
+using Microsoft.Extensions.Logging;
 using NServiceBus;
 
 namespace artiso.AdsdHotel.Black.Api
 {
     public class RequestGuestInformationHandler : IHandleMessages<RequestGuestInformation>
     {
+        private readonly IDataStoreClient dataStoreClient;
+        private readonly ILogger<RequestGuestInformationHandler> logger;
+
+        public RequestGuestInformationHandler(IDataStoreClient dataStoreClient, ILogger<RequestGuestInformationHandler> logger)
+        {
+            this.dataStoreClient = dataStoreClient;
+            this.logger = logger;
+        }
+
         public async Task Handle(RequestGuestInformation message, IMessageHandlerContext context)
         {
-            // Todo retrieve from storage
-            await context.Reply(new GuestInformationResponse {GuestInformation = new GuestInformation { FirstName = "test"} });
+            var result = await dataStoreClient.Get<GuestInformationRecord?>(r => r.OrderId == message.OrderId).ConfigureAwait(false);
+            // ToDo what should we return when nothing is found?
```

Comment:
i suggest just returning null

Reply:
I don't like `null`.

Reply:
I think we can return a static Empty result.

Reply:
The last project i had to work with Empty Results it became really annoying.  
null is the default so why not use it? 

But we should definetly talk about it in todays meeting

Reply:
We are going to work with a Wrapper Class